src/test/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/CaseServiceImplTest.java

```diff
@@ -86,10 +84,9 @@ public void getHHCaseByUPRNFound() throws Exception {
 
     CollectionCase hhCase = this.collectionCase.get(0);
 
-    List<CaseDTO> caseDTO = caseSvc.getHHCaseByUPRN(UPRN);
-    CaseDTO rmCase = caseDTO.get(0);
+    CaseDTO rmCase = caseSvc.getLatestValidNonHICaseByUPRN(UPRN);
 
-    assertThat(caseDTO, hasSize(1));
+    assertNotNull(rmCase);
```

Comment:
rename various test method names to match your new service method name

Reply:
It is also not clear to me that we have tested the max by createdDateTime functionality.  We need sample data with different created date times, and maybe an extra test method or 2  to ensure this is working as implemented.

Reply:
I just spotted this also Rob - no unit test coverage with more than case to assert the correct filtering. As there is duplicated functionality in the CaseServceImpl and an existing JIRA ticket to remove this duping, I have agreed with Andy that he will pick that other ticket up immediately after this one. In doing the de-dupe refactoring he can move tests that assert the same functionality in UACServiceImpl to test the de-duped method. Adding new unit tests now that will be removed is unnecessary

Reply:
Addressed HH issues and added unit tests for CaseServiceImpl -> tests latest and valid address and non HI cases